### DISCUSSION TITLE
Aktueller Kernel für 5590 / 7.50 Final verfügbar auf OSP

### DISCUSSION THREAD

**Discussion by maverick3005** [Freetz-NG/freetz-ng#788] [Unerfüllte Wünsche]
Gerade gesehen:

https://osp.avm.de/fritzbox/fritzbox-5590-fiber/

Endlich ist der aktuelle Kernel-Kram für die 5590 von AVM verfügbar,. nachem der vor wenigen Tagen angefordert wurde.
Für die 5530 kommt der auch in Kürze.

SO. und wie war des jetzt mit dem Diff erzeugen (ADv1xxx.file...) ? - wahrscheinlich den "AVM-trashbin" wieder erstmal cleanen und hoffen das sich halbwegs eine kernel-config, die von AVM verwendet wird, auch mit in den Paketen befindet....

Allerdings mangels Zugriff könnte ich sowieso das Diff auf keinen einzigen Mirror laden.
Mit nem PUSH-request alleine wärs da wohl nicht getan denke ich mal... ?

Entsprechende Basis-Kernelconfigs müssten ja ebenso noch erstellt werden.

Ferner müssten die FORCED defaults für Userland "DTC"+ "fitdump" und das forced loop-Modul wie bei der 7.30 wieder in die
Basis-config auch wieder mit rein. ggfs. auch eventuell noch einige BusyBox Optionen checken, falls zusätzlich noch was aktiviert sein muss.

**Comment by fda77** [comment]
Ja mach mal, ich habs hochgeladen

**Reply by maverick3005** [reply]
Wie jetzt ? Hast Du bereits einen Diff erstellt ?

Dann fehlen im Grunde noch die ganzen .img und .kos Teile, sowie die eigentliche Kernel-config für AVM als Diff. Der Teil wird
dann ja wieder ueber github automated erstellt, lokal fehlen mir die kompletten 30 GB teilweisen uralten images um da alle durchläufe erstmal testweise machen zu können.

**Reply by SaMMy-lacht** [reply]
Die ganzen Images, die freetz verwendet, sind nur 305 Objekte mit einer Größe von 7,3 GB (ohne inhaus und labor natürlich). Wie kommst du auf 30 GB

**Reply by fda77** [reply]
Dafür gibts doch ein Script. Am längsten hält der Download mit dieser beschissenen Telekom-Leitung auf ^^

**Comment by maverick3005** [comment]
@fda77 

Kannst Du bitte mal den hochgeladenen Kernel-Diff der 5590 7.50 überprüfen bitte ?
Gegen den passenden Vanilla-Kernel bzw. freetz lässt der sich nicht diffen ohne "HUNKs failed" und entsprechende Abbrüche....

**Reply by fda77** [reply]
erstell dir di Datei eben selbst wenn sie dir nicht passt

**Reply by maverick3005** [reply]
...das wird in diesem fall auch nicht anders gehen, sieht aus als avm wieder teile vergessen hat....

am besten mal 7.30 gpl mit 7.50 gpl vergleichen. wenn die sowas abliefern bringt keine automatic was, da musst eh wieder hand anlegen und ggfs. die fehlenden "header" oder "includes" aus nem anderen Paket mit heranziehen....

**Reply by fda77** [reply]
dann mach das mal

**Reply by maverick3005** [reply]
Nachdem wir zu zweit an den Sachen gesessen haben haben wir festgestellt das ausschliesslich mit der Adv1-Methode nur die 7.30 kompiliert werden kann. Der 7.50 Code verwendet "LiSi", es gibt einen kgw ordner vor dem kernel-tree, und die ganzen UAPI-Sachen werden jetzt bereits vorbereitend nach Kernel >5.x schon verwendet. Ferner sieht es auch so aus, als ob jetzt nur noch ein kernel
für alle Modelle verwendet wird und nicht mehr etliche Einzelkernel usammengeschustert sind.

Der KGW-Ordner enthählt eine sehr gute Dokumentation ueber extra Zwischschschritte im Bereich "UAPI" headers.

Dort sagt avm auch klar das zwingend zu erstellung des Release-Kernels minimum 5 extra Eviroments noch gesetzt sein müssen und zuerst in jedem Fall ein "olddefconfig" erfolgen muss, das mit sich alle benötigen UAPI-Header gesammelt an einem neuen ort befinden.

Das ist ausserhalnb unseres Scopes und wird ALLE künftigen Modelle mit Kernel 4.x betreffen mit Lab/Inhaus/Final bei 7.50,
da avm jetzt wohl alle Treiber zu allen Geräten in nur noch einem kernel drin hat.

Bei ddem pre-durchlauf uebergibt man dann auch die Ziel-Hardware. Das würde bedeuten das man künftig auch für Beta-versionen Kernel von anderen Boxen verwenden kann, da nach erster Durchsicht alle Treiber, inkl. PUMA sich bereits in den neuren kernels befinden.

Im neueren KGW dir sind sehr aktuelle und gute Dokumentationen enthalten wie man 100% compatible, neue Release Kernel
selber erstellen kann. Dazu muss das Basis-Framework der bisherigen ADv1-diff methode lediglich angepasst werden, das die neuen "Prefunktionen" zusätzlich noch eine Basis für die benötigte Hardware grund erstellt (so habe ich das jetzt nach der Doku verstanden).

Das liegt jetzt aber als komplette Neuerung ausserhalb unseres Scopes und da muss jemand beigehen der seit Jahren wirklich fitt
ist auf die AVM-Sachen. Valide kernel-basis configs hat AVM allerdings im 5590 7.50 paket mitgeliefert und die müssen auch zuerst einmal durchlaufen - und erst im Anschluss dann weitere Freetz-Optionen extra aktiviert werden, ggfs. muss dann LiSi/uapi zum update nochmal durchrennen.

**Comment by fda77** [comment]
Hä? Keine Ahnung was du meinst. Geh einfach Schritt nach Schritt der Anleitung vor

**Reply by maverick3005** [reply]
2 Leute haben genau das getan, es gibt änderungen bei den aktuellen kernels. Die vorhandene Methode ist ohne die neuen benötigten Enviroment sachen nicht mehr ausreichend. Daher reicht unser vorhandenes wissen hier nicht mehr.

Es wurde exakt die in freetz erwähnte methode angwendet und es gibt bei 7.30 keine LiSi/UAPI-Sachen, das ist jetzt NEU
und benötigt einen Pre-durchlauf und extra Enviroment-settings wohl noch, quasi einen Schritt mehr und das ist alles im KGW ordner auch
in deutsch beschrieben wie die exakte Reihenfolge sein soll einen RELEASE Kernel 1zu1 wie bei AVM nachbilden zu können.

Damals gabs diese Doku nicht. (kgw/doc steht das beschrieben im 5590 7.50 GPL teil)

Probiers doch selber mal Dein eigenes Paket für eine 5590 auf der final 7.50 zu backen, dann erklärt sich das für jemand wie Dich der sofort weiss, was da los ist, von selber, als das man das unadequat zu erläutern versucht.

Es bringt also nichts 2 Bäckern jetzt zum Schlachter zu machen wenn das erforderliche Basiswissen einfach nicht da ist in diesem Bereich.
Wir haben es VERSUCHT und nach Unterschieden gesucht. Und die habe ich bestmöglich mitgeteilt.

AVM hat dieses mal sogar keinen trashbin sondern tatsächlich eine lauffähige .conf abgeliefert wie ich das ersehen konnte...

**Comment by maverick3005** [comment]
so was kommt dann bei Deinem hochgeladenen ADv1 diff bei raus, ebenso wenn man sich ein ADv1 package lokal auch nochmal selber erstellt:

bei der 5590 7.50

make -f ./scripts/Makefile.headersinst obj=drivers/char/avm_new/include/uapi/avm/enh \
dst=drivers/char/avm_new/include/avm/enh \
gen=drivers/char/avm_new/include/uapi/avm/enh
make -f ./scripts/Makefile.headersinst obj=drivers/char/avm_new/include/uapi/avm/event \
dst=drivers/char/avm_new/include/avm/event \
gen=drivers/char/avm_new/include/uapi/avm/event
scripts/Makefile.headersinst:62: *** Missing generated UAPI file ./drivers/char/avm_new/include/uapi/avm/event/gen-types.h. Schluss.
make[4]: *** [avm/make/generated/genhdr-stub.make:62: __uapi_install/drivers/char/avm_new/include/uapi/avm/event/] Fehler 2
rm drivers/char/avm_new/include/uapi/avm/event/Kbuild drivers/char/avm_new/include/uapi/avm/enh/Kbuild
make[3]: *** [avm/make/generated/genhdr.make:45: __install_headers/drivers/char/avm_new] Fehler 2
make[2]: *** [GNUmakefile:143: headers_install] Fehler 2

Es fehlt der von AVM beschriebe "olddefconfig" zwingende schritt als erstes laut deren Hinweise, sowie auch die Definititon
von 5 Enviromentparmetern für den "Release-Kernel" evtl. noch. 

Der "event"-folder sowie etliche includes werden jetzt ueber das LiSI-Projekt/KGW an eine andere Stelle gepackt als bei der alten Methode, daher findet der derzeit implementierte Ablauf diverse Files nicht mehr an den Stellen, wo es bisher auch immer war offenbar ?

Die 7.29 zur 5590 lässt sich mit dem dazugehörigen ADv1 diff ebenfalls nicht mehr erstellen. Es geht tatsächlich nur der 7.30er GPL-code innerhalb von Freetz laut des Kollegen.

**Reply by fda77** [reply]
Okay, wenn es nicht mehr geht ist blöd

**Comment by maverick3005** [comment]
wie gesagt. wir haben es ja versucht. Jemand der das seit Jahren macht und einen Blick auf 7.30 und 7.50 eben wirft sieht sofort den unterschied. Aber allein das AVM jetzt wirklich mal einen sauberen GPL abgeliefert hat ohne "trashbin" und eine wirklich gute deutsche Doku im kgw/doc drin hat ist ein fortschritt.

Es wird dann wohl künftig offenbar nur noch EINEN Kernel geben für ALLE Plattformen. denn es sind auch sämtliche anderen Treiber mit in dieser Kernel-version. Das Mopsen einer subrevision von einer anderen Plattform wenn die noch kein GPL code released haben wird damit massiv einfacher. Ich habe sämtliche Hardware zu allen Boxen in der 7.50er version entdeckt wenn ich mich nicht irre.

Du könntest quasi ohne vanilla kernel hergehen dir einen 7.50er rauspicken und den als universalkernel fast schon verwenden so wie ich das sehe. so wie die das beschrieben haben mit dem olddefconfig und angabe der basisplattform erstellt er dann die basis für die Zielhardware schon und die 5 enviromentparameter machen dann sogar einen releasekernel der 1zu1 identisch sein soll, zum stock kernel.

Ich finde das neue System also schon vorteilhafter als das damalige inselgebastel. AVM scheint den Mainkernel jetzt auch ueber ein Repo intern zu pflegen.

Der Spass fängt aber erst bei den 4.4.1xx Kernels an. die ollen dinger die 4.4.60 usw. haben das noch nicht implementiert, allerdings die Betas und inhaus nutzen es schon. Einzig die 3.10er für die 7490 ist/bleibt auf uralt-stand wie gehabt.

**Comment by fda77** [comment]
Ich glaub dein Bekannter hat auch was falsch gemacht.
An dem Kernel ist nichts besonderes, übliche AVM "Qualität": Unpassende .config und dazu mies gepatcht:
```
net/sched/Kconfig:44: warning: menuconfig statement without prompt
arch/arm/Kconfig:1870:warning: defaults for choice values not supported
arch/arm/Kconfig:1878:warning: defaults for choice values not supported
arch/arm/Kconfig:1887:warning: defaults for choice values not supported
.config:502:warning: override: reassigning to symbol VMSPLIT_2G
warning: (QCOM_SYSMON && SAMPLE_QMI_CLIENT) selects QCOM_QMI_HELPERS which has unmet direct dependencies ((ARCH_QCOM || COMPILE_TEST) && NET && !ARCH_IPQ40XX && !ARCH_IPQ806x)

```

**Comment by maverick3005** [comment]
Puh. Das kann natürlich sein. Ich hatte am Wochenende keine Möglichkeit selber zu bauen, ausser den code mir am Laptop komplett durchzusichten und alles mal miteinander zu vergleichen.

Rennt das bei Dir innerhalb von freetz etwa ???

Mich haben diese neueren kgw-Sachen da schon irritiert und auch das da einige Verzeichnisse jetzt verschoben sind, wohl als Vorbereitung auf Kernel 5.x schon bei denen....


Fazit: es würde eine passende .config benötigt nebst der Reaktivierung der Kernelverfügbarkeit. Hatte da aber keinen Github-Durchlauf für die .kos Sachen gesehen die letzten 3 Tage, das da für den 7.50 eine saubere Basis "automatisch" erstellt wurde.

Es war ja zuerst einmal nur das ADv1-diff hochgeladen (vielen DANK!!)


FRAGE:
Könntest Du eine valide "config-qcaarmv8-5590_07.50.in" via github durchrennen lassen, das die sauber mit im index ist ?


Bei verschiedenen Modellen ist dann noch sowas aufgefallen:

In der Kernel-config: "5590--07_5X.in" steht:

```
config FREETZ_AVM_PROP_KERNEL_CONFIG_AREA_SIZE_0
        def_bool y
```
müsste da als wert nicht jetzt "_256" gesetzt sein? Der "fehler" ist auch bei der 7590 7.50 final noch drin. Hier ist es ebenfalls "0".

bei der 6890 ist es korrekt gesetzt mit 256. allerdings ist das keine fit-box. Ich bin mir nicht ganz sicher ob bei den FIT-Geräten,
die ebenfalls die neuen Kernel haben, der wert nicht auch angepasst werden müsste ?

Ich wüsste auf die Schnelle auch gar nicht wo man den korrekten Wert für jedes Gerät hernehmen sollte. Hat den avm mit in seiner mehr oder minder halbfertigen config schon drin stehen ?

**Reply by fda77** [reply]
Nachdem dein pushrequest gemergt ist rennt dass bei github durch

**Comment by maverick3005** [comment]
wohl klar. kannst Du auch mal die Sachen mit der kernel area size bei den verschiedenen Modellen nochmal durchsehen, das betrifft ja etliche Boxen ob das jetzt ueberall mit "0" so ok ist ?

Ich denke es scheitert gerade noch am lokalen durchlauf eine VALIDE config für den 7.50er 5590 zu bekommen, die muss ich ja auch irgendwo rauszerren, sonst hab ich nichts als request zum pushen. Zu viel Neuland auf einmal :-)

**Reply by fda77** [reply]
Unter welchem Punkt der Anleitung steht irgend was von "0" ?
Ich denke wir schieben dies lieber auf die Wishlist

**Reply by fda77** [reply]
Ausserdem deaboniere ich diese Diskussion. Den PR werd ich sehen

**Comment by fda77** [comment]
Der "Kollege" der für @maverick3005 programmiert ist wohl @SaMMy-lacht der im DEB prahlt dass er es ganz allein geschafft hat ein Modul zu bauen: https://www.digital-eliteboard.com/threads/522427/ Als Dank labert maverick/himan/loman ihn den ganzen lieben langen tag mit crypto-coin geschwafel voll. Auf die Idee dass sie was an freetz zurückgeben könnten kommt natürlich wieder keiner, Erbärmlich
